File: java/i00.abstract/Shape.java
public abstract class Shape {
   /******* FIELDS *******/
   protected Double width;
   protected Double height;

   /******* CONSTRUCTORS *******/
   public Shape(Double width, Double height)  {
     this.width = width;
     this.height = height;
   }

   public Shape(Integer width, Integer height)  {
     this.width  = new Double(width);   // convert to Double
     this.height = new Double(height);  // convert to Double
   }

   /******* METHODS *******/
   abstract public Double calculateArea();
}

File: cs/i00.abstract/Shape.cs
public abstract class Shape {
   /******* FIELDS *******/
   protected double width;
   protected double height;

   /******* CONSTRUCTORS *******/
   public Shape(double width, double height)  {
     this.width = width;
     this.height = height;
   }

   public Shape(int width, int height)  {
     this.width  = System.Convert.ToInt32(width);   // convert to double
     this.height = System.Convert.ToInt32(height);  // convert to double
   }

   /******* METHODS *******/
   public abstract double calculateArea();
}
